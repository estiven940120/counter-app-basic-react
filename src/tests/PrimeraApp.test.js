import React from 'react'
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';

import PrimeraApp from "../PrimeraApp";

describe('Pruebas en <PrimerApp />', () => {
    
    test('Debe mostrar Hola, mundo', () => {
       
        const saludo = "Hola, mundo";

        const wrapper = shallow( <PrimeraApp saludo={ saludo } />);
     
        expect( wrapper ).toMatchSnapshot();
    })

    test('Debe mostrar el subtitulo enviado por props', () => {
       
        const saludo = "Hola, mundo";
        const subtitulo = "Soy un subtitulo"
        const wrapper = shallow( 
                <PrimeraApp 
                    saludo={ saludo }
                    subtitulo={ subtitulo } 
                />
        );

        const textoParrafo = wrapper.find('p').text();
        console.log(textoParrafo);

        expect ( textoParrafo ).toBe( subtitulo );
    })
    
});