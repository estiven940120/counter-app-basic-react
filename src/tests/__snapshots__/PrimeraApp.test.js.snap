// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimerApp /> Debe mostrar Hola, mundo 1`] = `
<Fragment>
  <h1>
    Hola, mundo
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
